### ISSUE TITLE
Import/Export: API: Validation

### ISSUE THREAD

**Issue by kahboom** [syndesisio/syndesis#2081]
## This is a...
<pre><code>
[ x ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem
The UI needs to validate imported integrations, on two levels:
1. API. Server errors, invalid file format, etc.
2. UI. Immediate file type detection, size detection, etc. #1957

## Expected behavior
Display the appropriate error to the user in inline notifications.

## Screenshot
[Full Designs](https://github.com/syndesisio/syndesis/blob/master/ux/designs/importexport/importexport.md)

## Request and Response Data
Will fill this in once I have a better idea, but typically I would expect just a specific status integer (not just 200 and 500), with a string description of the response -- perhaps one that can be displayed to the user and another for debugging.

cc @deeleman - Did we agree on an interface for error validation from the API? Did anything ever come from that discussion?

## API Endpoints and Schemas

`/v1/integrations/import`

## Tasks involved / Steps to Reproduce

1. Navigate to Integrations > Import.
2. Do anything typically considered to be invalid (e.g. select the wrong file type).
3. No proper validation is shown to the user.

**Comment by deeleman** [comment]
Please bear in mind that you can an (and should) conduct file type validation from the Front-end

**Comment by kahboom** [comment]
Yes, this is already being done. File type validation is under the UI bullet point. The API example I listed is for proper "file format".

**Comment by gashcrumb** [comment]
So we need to clearly indicate that basically what this issue is for is for more fine-grained validation messages when the user uploads a file, other than just an HTTP 500 or HTTP 200.  And that this particular issue is for the server side, not the UI, we have #1957 for that.

Though if the system generates the export file, how much is the user going to be able to do with a more exact error message for the file import?  The zip file contents are opaque to the user, at least we should consider those contents to be so.  If anything more detailed information would be logged server-side.

**Comment by gashcrumb** [comment]
@chirino Thoughts on validation messages during import?
